src/commcare_cloud/ansible/roles/elasticsearch/tasks/main.yml

```diff
@@ -1,18 +1,45 @@
+- name: "Check if Cluster Name is updated for Elasticsearch V7"
+  assert:
+    that:
+      - "'es7_' in elasticsearch_cluster_name"
+    fail_msg: "Elasticsearch V7 cluster name should start with es7_ prefix"
+  when: elasticsearch_version  == "7.9.1"
+
 - name: Downloads Directory
   file: path="/home/{{ cchq_user }}/downloads/" state=directory
 
 - name: Create elasticsearch user
   user: name=elasticsearch createhome=no password=no state=present
 
-- name: Elasticsearch User
-  user: name=elasticsearch state=present
+- name: "Tune vm.max_map_count"
+  lineinfile:
+    dest: /etc/sysctl.conf
+    line: "vm.max_map_count=262144"
+    state: present
+  become: yes
 
-- name: Download Elasticsearch
+- name: "Tune vm.max_map_count (for current session)"
+  sysctl:
+    name: vm.max_map_count
+    value: '262144'
+    state: present
+  become: yes
+  
+- name: Download Elasticsearch 2.4.x
   become: yes
   get_url:
     url: "https://download.elasticsearch.org/elasticsearch/elasticsearch/elasticsearch-{{ elasticsearch_version }}.tar.gz"
     dest: "/home/{{ cchq_user }}/downloads/elasticsearch-{{ elasticsearch_version }}.tar.gz"
     sha256sum: "{{ elasticsearch_download_sha256 }}"
+  when: elasticsearch_version == "2.4.6"
+
+- name: Download Elasticsearch 7.9.1
```

Comment:
how about moving these downloads to the version specific task files (misc_v2 & misc_v7) (maybe rename them to es_v2 and es_v7

Reply:
The following tasks after this are common to both versions, moving it to version specific files require duplicating the common tasks in both files.